### PR TITLE
Added tests for xml parser source code and removed autogenerated useless code [10054]

### PR DESCRIPTION
**Pull request by rsanchez15**
This PR:
- Erase autogenerated useless code in `TCPControlMessage.cpp` and `TCPControlMessage.h`
- Separate dynamic XMLParser from std and add in several CMakeLists tests
- Adds remaining test cases for `XMLElementParser.cpp` to achive a 100% line coverage in that file.
- Adds remaining test cases for `XMLParser.cpp` to achive a 100% line coverage in that file.

### CODE REVIEW

File: test/unittest/xmlparser/XMLParserTests.cpp [hunk 1]
```diff
@@ -617,6 +661,2013 @@ TEST_F(XMLParserTests, DataBuffer)
...
+        EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::parseProfiles_wrapper(titleElement, *profilesNode));
+    }
+}
+
+/*
+ * This test checks the correct functioning of the parseProfile function for all unsupported xml child elements of
+ * <profile>.
+ * 1. Check that unsupported elements <qos_profile>, <application>, and <type> are read as xml child elements of the
+ * <profiles> root element.
+ * 2. Check that it outputs a Log Error when reading an unsupported element.
```
Comment (EduPonz) [inline test/unittest/xmlparser/XMLParserTests.cpp:2148]: Some tests do this check and some don't

**Review by EduPonz — CHANGES_REQUESTED** [review]
Partial review, Elements parser missing

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 2]
```diff
@@ -0,0 +1,3217 @@
...
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    // Check that the XML return code is correct for the publish mode kind setting.
+    EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::getXMLPublishModeQos_wrapper(titleElement, publishMode, ident));
+    // Check that the publish mode kind is set to SYNCHRONOUS_PUBLISH_MODE.
+    EXPECT_EQ(publishMode.kind, PublishModeQosPolicyKind::SYNCHRONOUS_PUBLISH_MODE);
+}
+
+/*
+ * This test checks the positive case of configuration via XML of the history memory policy dynamic reusable mode.
```
Comment (EduPonz) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:1578]: We could morph this test into check them all I think

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 3]
```diff
@@ -0,0 +1,3217 @@
...
+    // Invalid element
+    sprintf(xml, xml_p, "<bad_element> </bad_element>");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_ERROR,
+            XMLParserTest::getXMLRemoteLocatorsAllocationAttributes_wrapper(titleElement, allocation, ident));
+}
+
+/*
+ * This test checks the negative cases in the xml child element of <RemoteLocatorsAllocationAttributes>
```
Comment (EduPonz) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:2980]: I don't think this test is about allocation attributes

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 4]
```diff
@@ -0,0 +1,3217 @@
...
+        // Invalid argument
+        sprintf(xml, enum_p, "BAD FIELD");
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.RootElement();
+        EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLEnum_wrapper(titleElement, &e, ident));
+    }
+}
+
+/*
+ * This function is not implemented, so this test checks fulfill the XMLElementParser coverage
```
Comment (EduPonz) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:3082]:
```suggestion
 * This function is not implemented, so this test checks fulfillment of the XMLElementParser coverage
```

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 5]
```diff
@@ -0,0 +1,3217 @@
...
+ *      1. FILTER_DIFFERENT_PROCESS
+ */
+TEST_F(XMLParserTests, getXMLEnum_positive)
+{
+    uint8_t ident = 1;
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+    char xml[1000];
+
+    // IntraprocessDeliveryType Enum
```
Comment (EduPonz) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:3136]: I'd check all of them

**Review by EduPonz — CHANGES_REQUESTED** [review]
Second part of the review

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3201/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/415/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2866/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3233/)

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 6]
```diff
@@ -0,0 +1,3217 @@
...
+using eprosima::fastrtps::xmlparser::XMLP_ret;
+using eprosima::fastrtps::xmlparser::XMLParser;
+using eprosima::fastrtps::xmlparser::up_participant_t;
+using eprosima::fastrtps::xmlparser::up_node_participant_t;
+using eprosima::fastrtps::xmlparser::node_participant_t;
+using eprosima::fastrtps::xmlparser::sp_transport_t;
+
+using eprosima::fastrtps::xmlparser::XMLProfileManager;
+
+class XMLParserTests : public ::testing::Test
```
[the remark this answers is not in the document]
Comment (imontesino) [reply test/unittest/xmlparser/XMLElementParserTests.cpp:47]: - Moved XMLTreeTests to another file and executable. 
- Moved XMLParserTests class to separate header.

File: test/unittest/xmlparser/XMLParserTests.cpp (hunk 1 above)
Comment (imontesino) [reply test/unittest/xmlparser/XMLParserTests.cpp:2148]: Only test which produce log errors but do not return an error code. Only for unsupported tags where the rest of the XML is correct.

File: test/unittest/xmlparser/XMLParserTests.cpp [hunk 7]
```diff
@@ -617,6 +661,2013 @@ TEST_F(XMLParserTests, DataBuffer)
...
+        sprintf(xml, xml_p, e.c_str(), e.c_str());
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.RootElement();
+        profilesNode.reset(new BaseNode{ NodeType::PROFILES });
+        EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::parseProfiles_wrapper(titleElement, *profilesNode));
+    }
+
+    std::vector<std::string> elements_error {
+        "library_settings",
+        "participant",
```
[the remark this answers is not in the document]
Comment (imontesino) [reply test/unittest/xmlparser/XMLParserTests.cpp:2128]: The purpose of this part of the test was to force errors in each of the tags, not for the tags themselves to be not valid profile types.

File: test/unittest/xmlparser/XMLParserTests.cpp [hunk 8]
```diff
@@ -617,6 +661,2013 @@ TEST_F(XMLParserTests, DataBuffer)
...
+}
+
+/*
+ * This test checks all possible settings of the TLS verify mode (<verify_mode>) that return an XML OK code.
+ * 1. Check that the VERIFY_NONE setting return an xml ok code and is set correctly.
+ * 2. Check that the VERIFY_PEER setting return an xml ok code and is set correctly.
+ * 3. Check that the VERIFY_FAIL_IF_NO_PEER_CERT setting return an xml ok code and is set correctly.
+ * 4. Check that the VERIFY_CLIENT_ONCE setting return an xml ok code and is set correctly.
+ */
+TEST_F(XMLParserTests, parseTLSConfigVerifyMode)
```
[the remark this answers is not in the document]
Comment (imontesino) [reply test/unittest/xmlparser/XMLParserTests.cpp:1862]: `TEST_F` does not allow dots in the test names, subtypes will be separated by an underscore

File: test/unittest/xmlparser/mock/XMLMockConsumer.h [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+    virtual void Consume(
+            const Log::Entry& entry)
+    {
+        std::unique_lock<std::mutex> guard(*mutex_);
+        entries_consumed_.push_back(entry);
+        cv_.notify_one();
+    }
+
+    std::vector<Log::Entry> ConsumedEntries() const
```
[the remark this answers is not in the document]
Comment (imontesino) [reply test/unittest/xmlparser/mock/XMLMockConsumer.h:45]: Parent class is styled in cammel case.

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 10]
```diff
@@ -0,0 +1,3217 @@
...
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLRemoteServer_wrapper(titleElement, attr, ident));
+}
+
+/*
+ * This test checks the negative cases of a <port> xml element.
+ * 1. Check a missing case of each of the <port> child tags.
+ * 2. Check a wrong child tag.
+ */
+TEST_F(XMLParserTests, getXMLPortParametersNegativeClauses)
```
[the remark this answers is not in the document]
Comment (imontesino) [reply test/unittest/xmlparser/XMLElementParserTests.cpp:705]: These are the negative clauses.

File: include/fastdds/rtps/transport/tcp/TCPControlMessage.h [hunk 11]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (EduPonz) [inline include/fastdds/rtps/transport/tcp/TCPControlMessage.h:1]: Existing files should keep their license data, new files should have the new one

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — DISMISSED** [review]
LGTM! Waiting on CI :mechanical_arm:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3237/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/448/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2894/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3269/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3241/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/452/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2898/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3273/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3255/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/466/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2907/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3287/)

**Review by EduPonz — APPROVED** [review]
LGTM, all failures seem unrelated

**Pull request merged**